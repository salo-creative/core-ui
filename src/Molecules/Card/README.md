# Card

Card is a pure styled component render. Renders a standard `<div>` with themed styling for a card

-------

## Usage

Install

```javascript
yarn add @salo/core-ui
```

Include

```javascript
import Card from '@salo/core-ui/Atoms/Card';
```

Implement as follows

```javascript
<Card>
  // My content
</Card>
```

### For full prop types and usage see storybook info/knobs