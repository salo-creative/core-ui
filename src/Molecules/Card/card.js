import React from 'react';
import PropTypes from 'prop-types';

// COMPONENTS & STYLES
import { CardWrapper } from './card.styles';

// HELPERS & CONSTANTS
import { colours } from '../../helpers/colours';

const Card = (props) => {
  const { background, borderRadius, children, margin, padding } = props;

  return (
    <CardWrapper
      background={ background }
      borderRadius={ borderRadius }
      margin={ margin }
      padding={ padding }
    >
      { children }
    </CardWrapper>
  );
};

Card.defaultProps = {
  background: colours.grey,
  borderRadius: '0.4rem',
  children: 'No content supplied!',
  margin: '0',
  padding: '2rem'
};

Card.propTypes = {
  background: PropTypes.string,
  borderRadius: PropTypes.string,
  children: PropTypes.any,
  margin: PropTypes.string,
  padding: PropTypes.string
};
export default Card;