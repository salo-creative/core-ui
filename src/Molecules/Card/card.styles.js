import styled from 'styled-components';

// HELPERS
import { boxShadow } from '../../helpers/colours';

export const CardWrapper = styled.div`
  background: ${ ({ background }) => background };
  border-radius: ${ ({ borderRadius }) => borderRadius };
  box-shadow: ${ boxShadow() };
  margin: ${ ({ margin }) => margin };
  overflow: hidden;
  padding: ${ ({ padding }) => padding };
  width: 100%;
`;