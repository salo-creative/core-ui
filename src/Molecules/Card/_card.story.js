import React from 'react';
import { storiesOf } from '@storybook/react';

// load tests
import { withTests } from '@storybook/addon-jest';
import { withKnobs, text, boolean } from '@storybook/addon-knobs';
import results from '../../../.storybook/jest-test-results.json';

// FEATURED COMPONENT
import {
  Card, colours, FlyOut, FlyOutButton, FlyOutLink
} from '../../index';

// README //
import README from './README.md';

// Start of story logic
const stories = storiesOf('Molecules | Card', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests({
  results
}));
stories.addParameters({
  jest: ['card']
});

stories.add(
  'Basic',
  (() => {
    const background = text('background', colours.grey);
    const borderRadius = text('borderRadius', '0.4rem');
    const margin = text('margin', '0');
    const padding = text('padding', '2rem');
    return (
      <Card
        background={ background }
        borderRadius={ borderRadius }
        margin={ margin }
        padding={ padding }
      >
        something
      </Card>
    );
  }), {
    notes: README
  }
);

stories.add(
  'Action',
  (() => {
    const background = text('background', colours.grey);
    const borderRadius = text('borderRadius', '0.4rem');
    const margin = text('margin', '0');
    const padding = text('padding', '2rem');
    const cardActionsTop = boolean('cardActionsTop', true);
    return (
      <Card
        background={ background }
        borderRadius={ borderRadius }
        margin={ margin }
        padding={ padding }
      >
        something<br />more<br />more<br />more<br />more<br />more<br />more<br />more<br />more<br />more<br />more<br />more<br />more<br />more<br />more<br />more<br />
        <FlyOut context='card' cardActionsTop={ cardActionsTop }>
          <FlyOutButton title='Title' onClick={ () => alert('Button Clicked') } icon='sync' />
          <FlyOutLink title='Title' link='#' icon='dashboard' />
        </FlyOut>
      </Card>
    );
  }), {
    notes: README
  }
);