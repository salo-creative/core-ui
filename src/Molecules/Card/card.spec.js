import React from 'react';
import 'jest-styled-components';

import { renderWithTheme } from '../../../test';

// COMPONENT
import Card from './index';

test('Renders against snapshot', async () => {
  const { container } = renderWithTheme(
    <Card />
  );
  expect(container.firstChild).toMatchSnapshot();
});