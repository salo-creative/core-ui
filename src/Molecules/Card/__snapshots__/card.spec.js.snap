// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders against snapshot 1`] = `
.c0 {
  background: #e9ecef;
  border-radius: 0.4rem;
  box-shadow: 2px 2px 2px 0px rgba(0,0,0,0.2);
  margin: 0;
  overflow: hidden;
  padding: 2rem;
  width: 100%;
}

<div
  class="c0"
>
  No content supplied!
</div>
`;
